techdebt: [CDS-82834]: Revamp Connect and Manage Argo Providers - technical changes (#4336)
harness/developer-hub: docs/continuous-delivery/gitops/connect-and-manage/multiple-argo-to-single-harness.md

---

> title: Bring Your Own Argo CD (BYOA) into Harness GitOps

Author: done

  - /docs/continuous-delivery/gitops/multiple-argo-to-single-harness
---

> In this topic, learn how to manage a Bring Your Own Argo CD (BYOA) setup in Harness.

Author: done

    serviceaccount/byoa-agent-agent-upgrader created  
    cronjob.batch/byoa-agent-agent-upgrader created
    ```
12. In the Harness GitOps Agent installer, select **Continue**.
   
   The Agent has registered with Harness.
   
   ![](./static/multiple-argo-to-single-harness-67.png)

13.  Select **Continue**. The **Map Projects** settings appear.

:::warning

> Make sure that you deploy only one GitOps Agent per Argo CD namespace. Deploying multiple Agents created in different projects or accounts can lead to unpredictable behaviour.

Reviewer: FYI - we use American English spelling, so no "our" etc.

Harness supports mapping Argo CD projects into Harness projects through the Agent that controls the Argo CD deployment. When importing Argo CD projects, Harness maps Argo CD projects into Harness projects that belong to one account.

Enabling multiple Agents in one Argo CD namespace implies cross-account resource sharing which Harness does not support.

:::

## Map existing Argo projects

> To map your existing Argo CD projects to Harness projects, select the Argo CD projects you want to use, and then select the Harness project to which you want to map the Argo CD project.

Reviewer: ```suggestion
To map your existing Argo CD projects to Harness projects, select the Argo CD projects you want to use, and then select the Harness project where you want to map the Argo CD project.
```

1. In **Map Projects**, in **Select your Argo Projects to Import**, select the Argo CD projects you want to map.
   
   ![](./static/multiple-argo-to-single-harness-68.png)

2. In each row, select the Argo project and the Harness project.
   
   ![](./static/multiple-argo-to-single-harness-69.png)
   
   :::note
   Do not map the same Argo CD project to multiple Harness projects.
   :::

3. Select **Import & Continue**.
   
   The Argo CD projects are imported.

>    Here's an example in which the Argo CD applications, repositories, repository certs, and clusters are imported.

Reviewer: ```suggestion
   Here's an example where the Argo CD applications, repositories, repository certs, and clusters are imported.
```

Reviewer: our style is informal, so we use "where" instead of "in which".

   ![](./static/multiple-argo-to-single-harness-70.png)

4. Select **Finish**. The mapping is displayed in the Agent details.
   
   ![](./static/multiple-argo-to-single-harness-71.png)
   
5. Select **Save**.

## Verify mapping

> Next, look in the mapped Harness project to see the imported Argo CD entities.

Reviewer: ```suggestion
To see the imported Argo CD entities from the mapping, look in the mapped Harness project.
```

   
   ![](./static/multiple-argo-to-single-harness-73.png)
   
   Let's add a new mapping.

3. In **Mapped Harness Project**, select **Add**.
4. In **Map Projects**, in **Select your Argo Projects to Import**, select an Argo CD project. Do not select a project you have already mapped.
5. Map the Argo CD project to a Harness project and select **Import & Continue**.
   Do not re-map an existing mapping. Harness will throw an error.
   
   ![](./static/multiple-argo-to-single-harness-74.png)

6. When the import is complete, select **Finish**.

Both projects are now mapped.

![](./static/multiple-argo-to-single-harness-75.png)

## Adding Argo CD entities to Harness automatically

When an Agent contains Argo CD project mappings to a Harness project, any new entities added to the Argo CD project are added to the mapped Harness project automatically.

Try adding a new Argo CD repository to the mapped Argo CD project.

Once it is saved in Argo CD, go to Harness and look at the GitOps repositories in the mapped Harness project. A new repository is added.

### Automatically added Argo CD repositories

> When Argo CD repositories are added to Harness, Harness automatically generates a name for the repository. This is necessary because Argo CD has no name setting for its repositories. To generate a name for a repository, Harness remove any dashes from the name of the repository, and then appends an underscore followed by a unique suffix.

Reviewer: ```suggestion
When Argo CD repositories are added to Harness, Harness automatically generates a name for each repository. Name generation is necessary because Argo CD has no name setting for its repositories. When generating a name for a repository, Harness removes any dashes from the name of the repository, and then appends an underscore followed by a unique suffix.
```

For example, the Argo CD repository `https://github.com/argoproj/gitops-engine.git` is named `gitopsengine_kmjzyrbs` in Harness.

![](./static/multiple-argo-to-single-harness-76.png)

> ### Make sure that your Argo CD entities are visible in Harness

Reviewer: ```suggestion
### Ensure your Argo CD entities are visible in Harness
```
